proxyclient/m1n1/hv/__init__.py

```diff
@@ -1487,51 +1487,9 @@ def enable_time_stealing(self):
     def disable_time_stealing(self):
         self.p.hv_set_time_stealing(False)
 
-    def load_macho(self, data, symfile=None):
-        if isinstance(data, str):
-            data = open(data, "rb")
-
-        self.macho = macho = MachO(data)
-        if symfile is not None:
-            if isinstance(symfile, str):
-                symfile = open(symfile, "rb")
-            syms = MachO(symfile)
-            macho.add_symbols("com.apple.kernel", syms)
-            self.xnu_mode = True
-
-        self.symbol_dict = macho.symbols
-        self.symbols = [(v, k) for k, v in macho.symbols.items()]
-        self.symbols.sort()
-
-        def load_hook(data, segname, size, fileoff, dest):
-            if segname != "__TEXT_EXEC":
-                return data
-
-            print(f"Patching segment {segname}...")
-
-            a = array.array("I", data)
-
-            output = []
-
-            p = 0
-            while (p := data.find(b"\x20\x00", p)) != -1:
-                if (p & 3) != 2:
-                    p += 1
-                    continue
-
-                opcode = a[p // 4]
-                inst = self.hvc((opcode & 0xffff))
-                off = fileoff + (p & ~3)
-                if off >= 0xbfcfc0:
-                    print(f"  0x{off:x}: 0x{opcode:04x} -> hvc 0x{opcode:x} (0x{inst:x})")
-                    a[p // 4] = inst
-                p += 4
-
-            print("Done.")
-            return a.tobytes()
-
-        #image = macho.prepare_image(load_hook)
-        image = macho.prepare_image()
+    #only meant to be called from load_raw or load_macho
```

Comment:
Style: please leave a blank line between methods (I know we're not 100% consistent with this, but especially for complex classes like this one it helps).

Reply:
resolved with the refactor